src/contractors.jl

```diff
@@ -23,38 +29,54 @@ function (contractor::Bisection)(X, tol)
     return :unknown, X
 end
 
-# Newton
-struct Newton{F,FP} <: Contractor{F}
-    f::F
-    f′::FP   # use \prime<TAB> for ′
-end
+doc"""
+    newtonlike_contract(op, X, tol)
 
-Base.isinf(X::IntervalBox) = any(isinf.(X))
+    Contraction operation for contractors using the first derivative of the
+    function. This contraction use a bisection scheme to refine the intervals
+    with `:unkown` status.
 
-function (C::Newton)(X, tol)
+    Currently `Newton` and `Krawczyk` contractors uses this.
+"""
+function newtonlike_contract(op, C, X, tol)
```

Comment:
I am not sure this solution is ideal. However I feel like the most natural way of doing this would be to not have the contractors callable, but to have a method

```julia
contract(C::Contractor, X, tol)
```

and to use multiple dispatch/generated functions to handle everything correctly. Since it would be a deeper design change I wanted to discuss it first and it may be better in a separate PR.

Reply:
That definitely sounds like an interesting idea. I'm not sure exactly what it would gain - I guess that's why a PR would be useful.

Reply:
I feel that it's more natural to call a contractor `C` as `C(X)` to reflect the mathematical structure. This is just as possible to dispatch on, no? Or what do you gain by introducing the `contract` function?

Reply:
Thanks, yeah I was trying to do what you're suggesting and ran into the same frustrating problem :/ 
Let's leave it like this for the moment.